### PR TITLE
fix(buildme.sh): Fixes SunOS syntax errors & matches so and pm versions.

### PR DESCRIPTION
**Pull request by fsbruva**
This corrects some errors introduced in #31, where certain "if" statements
always evaluated as true, regardless of conditions. It also consolidates and
streamlines the replication of .pm files, and thus corrects #35. Also removes
execute flags from patch.

### CODE REVIEW

File: CPAN/buildme.sh [hunk 1]
```diff
@@ -1575,13 +1563,7 @@ if [ $PERL_MINOR_VER -ge 12 ]; then
     ARCH=`$PERL_BIN -MConfig -le 'print $Config{archname}' | sed 's/gnu-//' | sed 's/^i[3456]86-/i386-/' | sed 's/armv.*?-/arm-/' `
 fi
 mkdir -p $PERL_ARCH/$ARCH
-rsync -amv --include='*/' --include='*.so' --include='*.bundle' --include='autosplit.ix' --exclude='*' $PERL_BASE/lib/perl5/*/auto $PERL_ARCH/$ARCH/
-if [ -n "${buildIOSocketSSL}"  ]; then
-    rsync -amv --include='*/' --include='Socket/' --include='*.pm' --exclude='*' $PERL_BASE/lib/perl5/IO $PERL_ARCH/$ARCH/
-    rsync -amv --include='*/' --include='IDN' --include='*.pm' --exclude='*' $PERL_BASE/lib/perl5/Net $PERL_ARCH/$ARCH/
```
Comment (mherger) [inline CPAN/buildme.sh:1581]: Don't know what this item "IDN" is. But are you sure it's covered by your single-liner?

Comment (fsbruva) [reply CPAN/buildme.sh:1581]: Woops. You're right. Corrected with a rebase. Now everything within perl5/lib is replicated into $PERL_ARCH/$ARCH.

Comment (mherger) [reply CPAN/buildme.sh:1581]: I don't think your change made it to the pull request?

File: CPAN/buildme.sh [hunk 2]
```diff
@@ -1575,13 +1563,8 @@ if [ $PERL_MINOR_VER -ge 12 ]; then
...
-    rsync -amv --include='*/' --include='*.pm' --exclude='*' $PERL_BASE/lib/perl5/$ARCH/Net $PERL_ARCH/$ARCH/
-    rsync -amv --include='*/' --include='*.al' --include='autosplit.ix' --exclude='*' $PERL_BASE/lib/perl5/*/auto $PERL_ARCH/$ARCH/
-fi
+rsync -amv --include='*/' --include='*.so' --include='*.bundle' --include='autosplit.ix' --include='*.pm' --include='*.al' --exclude='*' $PERL_BASE/lib/perl5/$ARCH $PERL_ARCH/
+rsync -amv --exclude=$ARCH --include='*/' --include='*.so' --include='*.bundle' --include='autosplit.ix' --include='*.pm' --include='*.al' --exclude='*' $PERL_BASE/lib/perl5/ $PERL_ARCH/$ARCH/
```
Comment (fsbruva) [inline CPAN/buildme.sh:1567]: This line (1567) was added to grab everything else in perl5/lib. I rebased, so I didn't clog up the logs with two commits when it should have been one.

**Comment by mherger** [comment]
Ok, let's give this a try. Thanks!

**Pull request merged**